repo/fsrepo/fsrepo.go

```diff
@@ -379,6 +398,61 @@ func (r *FSRepo) openDatastore() error {
 	return nil
 }
 
+var SpecFn = "spec"
+
+func (r *FSRepo) readSpec() (string, map[string]interface{}, error) {
+	fn, err := config.Path(r.path, SpecFn)
+	if err != nil {
+		return "", nil, err
+	}
+	b, err := ioutil.ReadFile(fn)
+	if err != nil {
+		return "", nil, err
+	}
+	idspec := make(map[string]interface{})
+	err = json.Unmarshal(b, &idspec)
+	if err != nil {
+		return "", nil, err
+	}
+	id, ok := idspec["id"].(string)
+	if !ok {
+		return "", nil, fmt.Errorf("could not retrieve 'id' field from spec file")
+	}
+	spec, ok := idspec["spec"].(map[string]interface{})
+	if !ok {
+		return "", nil, fmt.Errorf("could not retrieve 'spec' field from spec file")
+	}
+
+	dsc, err := AnyDatastoreConfig(spec)
+	if err != nil {
+		return "", nil, err
+	}
+	computedId := dsc.DiskId()
+	if computedId != id {
+		return "", nil, fmt.Errorf("bad spec file, computed id (%s) does not match given (%s)",
+			computedId, id)
+	}
+
+	return id, spec, nil
+}
+
+func (r *FSRepo) writeSpec(id string, spec map[string]interface{}) error {
+	fn, err := config.Path(r.path, SpecFn)
+	if err != nil {
+		return err
+	}
+	idspec := map[string]interface{}{
+		"id":   id,
+		"spec": spec,
+	}
+	b, err := json.Marshal(idspec)
+	err = ioutil.WriteFile(fn, b, 0666)
```

Comment:
While the database directory shouldn't be world readable so this may not make a difference, this file really shouldn't be world writable.